[FIX] Add delay time to sync button when buying seeds and tools

DeMouraZ opened the pull request:
# Description

Added a little delay to the sync button which appeared too fast making lots of users to hit it and get metamask popup.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Manual test + yarn test

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

Polysemouse commented:
I think the rusty shovel / sand shovel / magic seeds or any other shops that has the sync button should have this new component

DeMouraZ commented:
Thanks for comment Polymouse, I will update with the rusty shovel and Magic Seeds on Helios, also the Sand Shovel in Treasure Island

DeMouraZ commented:
Updated following Polymouse comment, implemented on Rusty Shovel, Magic Beans and Sand Shovel.

The sync phrase changes from place to place, some of the examples I believe because of the UI perspective.
I left the original from the Seeds modal - Sync your farm to the Blockchain to restock
The shortest version would be - Sync your farm on chain to restock
If you guys want me to change to the smallest one just tell me, I didn't saw any problems with the UI just in some cases it occupates 2 lines if playing with mobile.

c-py reviewed (approved):
Awesome contribution! This pesky Sync button has got me numerous times 😅 

I noticed that when browsing the shop, they delay still happens. What do you think about adding `stock` as a property to prevent this? It would be possible to conditionally disable the button based on this input.

``` js
useEffect(() => {
  if (stock === 0 && !isLoaded) {
    setIsDisabled(true);
  }

  if (stock === 0 && isLoaded) {
    const timer = setTimeout(() => {
      setIsDisabled(false);
    }, 500);
    return () => clearTimeout(timer);
  }

  setIsLoaded(true)
}, [stock]);

  ```

c-py commented on src/features/island/buildings/components/building/market/Delayed.tsx:
```diff
@@ -0,0 +1,28 @@
+      setIsDisabled(false);
+    }, 500);
+    return () => clearTimeout(timer);
+  });
```
```suggestion
  }, []);
```

Supplying an empty array as the dependencies will stop this from running more than once.

c-py replied:
```suggestion
  }, []);
```

Supplying an empty array as the dependencies will stop this from running more than once.

DeMouraZ commented:
Thanks @c-py for approving my first PR, I hope that is one of many.

I've seen the issue when browsing on all the different modals, tried your sollution and others as well but I'm fairly new to this JS/React/Typescript scene and couldn't figure it out.

I sent you an friend invitation on discord (BallersOut) so we could discuss it better.
